UI/CMakeLists.txt

```diff
@@ -97,8 +97,11 @@ endif()
 foreach(graphics_library IN ITEMS opengl metal d3d11)
   string(TOUPPER ${graphics_library} graphics_library_U)
   if(TARGET OBS::libobs-${graphics_library})
-    target_compile_definitions(obs-studio
-                               PRIVATE DL_${graphics_library_U}="$<TARGET_FILE_NAME:OBS::libobs-${graphics_library}>")
+    target_compile_definitions(
+      obs-studio
+      PRIVATE
+        DL_${graphics_library_U}="$<$<IF:$<BOOL:${OS_LINUX}>,TARGET_SONAME_FILE_NAME,TARGET_FILE_NAME>:OBS::libobs-${graphics_library}>"
```

Comment:
Is this change necessary to get the _versioned_ name of the library set as a define?

Reply:
To get the SONAME one only on Linux yes. `TARGET_FILE_NAME` will put `libobs-opengl.so.30.1.0` where we need the SONAME (`libobs-opengl.so.30`) given by `TARGET_SONAME_FILE_NAME`.

Otherwise, we just create the same breakage we filename reversed.

Reply:
I'd check if just using `TARGET_SONAME_FILE_NAME` for all platforms shouldn't work out of the box - we set `NO_SONAME` for macOS and Windows doesn't support that anyway, so it would be interesting to see if that generator expression falls back on the target name then.

Otherwise I'd be fine using a conditional branch here instead (you can compose a string to _contain_ a generator expression, it doesn't need to be used at the immediate call site).

Reply:
> I'd check if just using `TARGET_SONAME_FILE_NAME` for all platforms shouldn't work out of the box - we set `NO_SONAME` for macOS and Windows doesn't support that anyway, so it would be interesting to see if that generator expression falls back on the target name then.

I'm only fixing the Linux path, I don't plan to test CMake behavior on other platform.

> Otherwise I'd be fine using a conditional branch here instead (you can compose a string to _contain_ a generator expression, it doesn't need to be used at the immediate call site).

I preferred to keep it simple and avoid duplicating code and/or effort in the three platform file.

Reply:
We cannot just fix a thing for one platform, we always have to code with _all_ supported platforms in mind, particularly for code that effects all platforms (in this case `libobs-opengl` as it's not a platform-specific target).

Reply:
Changed the logic to use `PLATFORM_ID`